diff --git a/macros/run.mac b/macros/run.mac
index 1294dd0..a7e6ccf 100644
--- a/macros/run.mac
+++ b/macros/run.mac
@@ -2,4 +2,4 @@
 /run/initialize
 /mygen/sourceType neutron
 /run/printProgress 1000000
-/run/beamOn 20000
+/run/beamOn 1000000
diff --git a/src/PhysicsList.cc b/src/PhysicsList.cc
index 8322942..433ed56 100644
--- a/src/PhysicsList.cc
+++ b/src/PhysicsList.cc
@@ -7,7 +7,7 @@ PhysicsList::PhysicsList(): FTFP_BERT_HP() //親クラスのコンストラク
 {
     RegisterPhysics(new G4RadioactiveDecayPhysics());
     //RegisterPhysics(new G4DecayPhysics()); <-- これはFTFP_BERT_HPに含まれているので、重複して登録するとエラーになる
-    RegisterPhysics(new G4EmLivermorePhysics());
+    ReplacePhysics(new G4EmLivermorePhysics());
     RegisterPhysics(new G4OpticalPhysics());
 }
 
diff --git a/src/RunConfig.cc b/src/RunConfig.cc
index 6250c3e..8902d7d 100644
--- a/src/RunConfig.cc
+++ b/src/RunConfig.cc
@@ -21,6 +21,7 @@ void RunConfig::SetSourceType(const G4String& sourceType)
     }
     */
 
+    
     if (sourceType == "neutron") {
         fParticleName = "neutron";
         fEnergy = 1.00 * MeV;
